packages/superset-ui-control-utils/src/types.ts

```diff
@@ -228,34 +233,42 @@ export type SharedSectionAlias =
   | 'sqlaTimeSeries'
   | 'NVD3TimeSeries';
 
-export interface ControlItem {
+export interface OverrideSharedControlItem {
   name: SharedControlAlias;
-  config: Partial<ControlConfig>;
+  override: Partial<SharedControlConfig>;
 }
 
-export interface CustomControlItem {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface CustomControlItem<P = any> {
   name: string;
-  config: ControlConfig;
+  config: CustomComponentControlConfig<P>;
 }
 
-export type ControlSetItem = SharedControlAlias | ControlItem | CustomControlItem | ReactNode;
+export type ControlSetItem =
+  | SharedControlAlias
+  | OverrideSharedControlItem
+  | CustomControlItem
+  | ReactElement
+  | null;
+
 export type ControlSetRow = ControlSetItem[];
 
 // Ref:
 //  - superset-frontend/src/explore/components/ControlPanelsContainer.jsx
 //  - superset-frontend/src/explore/components/ControlPanelSection.jsx
 export interface ControlPanelSectionConfig {
   label: ReactNode;
-  controlSetRows: ControlSetRow[];
   description?: ReactNode;
   expanded?: boolean;
   tabOverride?: TabOverride;
+  controlSetRows: ControlSetRow[];
```

Comment:
Is this intentional? Seems like just changing order.

Reply:
Yes, it's just changing order. `controlSetRows` is a larger object. Kind of feel it should be at the bottom because of the "weight".